cmd/otel-agent/main.go

```diff
@@ -17,7 +17,13 @@ import (
 	_ "github.com/DataDog/datadog-agent/pkg/version"
 )
 
+var BYOC string
```

Comment:
I would maybe move this to `cmd.otel-agent.command.BYOC` and just have it next to `cmd.otel-agent.command.loggerName`, and keep the `MakeRootCommand` prototype unchanged. You can then access the variable directly in the package. 

Also while this **has** to be a string, because it's a build-time variable, all downstream calls should use a bool.